File: 240410/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.py
N = int(input())

class person:
    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight

people = []
for _ in range(N):
    name, height, weight = input().split()
    height, weight = int(height), int(weight)
    people.append(person(name, height, weight))

people.sort(key = lambda x: (x.height, -x.weight))

for person in people:
    print(person.name, person.height, person.weight)

File: 260105/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

struct Person
{
    public:
        string name;
        int height;
        int weight;
};

int main() 
{
    int N;
    cin >> N;

    vector<Person> people;
    people.reserve(N);

    for(int i = 0; i < N; i++)
    {
        string name;
        int height, weight;
        cin >> name >> height >> weight;
        people.push_back({name, height, weight});
    }

    // Please write your code here.
    sort(people.begin(), people.end(), 
        [](const Person& a, const Person& b)
            {
                if(a.height != b.height)
                    return a.height < b.height;
                return a.weight > b.weight;
            });

    for(const auto& p: people)
    {
        cout << p.name << " " << p.height << " " << p.weight << endl;
    }
    return 0;
}